services-camel/src/test/java/edu/unc/lib/dl/services/camel/fulltext/FulltextProcessorTest.java

```diff
@@ -16,133 +16,79 @@
 
-import java.io.BufferedReader;
 import java.io.BufferedWriter;
 import java.io.File;
 import java.io.FileWriter;
-import java.io.InputStream;
-import java.io.InputStreamReader;
-import java.util.stream.Collectors;
 
 import org.apache.camel.Exchange;
 import org.apache.camel.Message;
+import org.apache.commons.io.FileUtils;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
-import org.mockito.ArgumentCaptor;
-import org.mockito.Captor;
+import org.junit.rules.TemporaryFolder;
 import org.mockito.Mock;
 
-import edu.unc.lib.dl.fcrepo4.BinaryObject;
-import edu.unc.lib.dl.fcrepo4.FileObject;
-import edu.unc.lib.dl.fcrepo4.RepositoryObjectLoader;
-import edu.unc.lib.dl.fedora.PID;
-import edu.unc.lib.dl.rdf.PcdmUse;
-import edu.unc.lib.dl.services.camel.fulltext.FulltextProcessor;
-
 public class FulltextProcessorTest {
     private FulltextProcessor processor;
-    private final String slug = "full_text";
-    private final String fileName = "full_text.txt";
+    private final String originalFileName = "full_text.txt";
     private final String testText = "Test text, see if it can be extracted.";
-    private int maxRetries = 3;
-    private long retryDelay = 10;
-    private File file;
-    private final static String MIMETYPE = "text/plain";
+    private final String derivativeFinalPath = "de/75/d8/11/de75d811-9e0f-4b1f-8631-2060ab3580cc";
+    private File originalFile;
+    private String derivPath;
     private final static String BINARY_URI =
             "http://fedora/content/45/66/76/67/45667667-ed3f-41fc-94cc-7764fc266075/datafs/original_file";
+    private static final String FEDORA_BASE = "http://example.com/rest/";
 
-    @Mock
-    private BinaryObject binary;
-    @Mock
-    private FileObject parent;
+    private static final String RESC_ID = FEDORA_BASE + "content/de/75/d8/11/de75d811-9e0f-4b1f-8631-2060ab3580cc";
 
-    @Mock
-    private RepositoryObjectLoader repoObjLoader;
+    @Rule
+    public TemporaryFolder tmpDir = new TemporaryFolder();
 
     @Mock
     private Exchange exchange;
 
     @Mock
     private Message message;
 
-    @Captor
-    private ArgumentCaptor<InputStream> inputStreamCaptor;
-
     @Before
     public void init() throws Exception {
         initMocks(this);
-        processor = new FulltextProcessor(repoObjLoader, slug, fileName, maxRetries, retryDelay);
-        file = File.createTempFile(fileName, "txt");
-        file.deleteOnExit();
-        when(exchange.getIn()).thenReturn(message);
 
-        when(repoObjLoader.getBinaryObject(any(PID.class))).thenReturn(binary);
+        derivPath = tmpDir.newFolder().getAbsolutePath();
+        processor = new FulltextProcessor(derivPath);
+        originalFile = tmpDir.newFile(originalFileName);
 
+        when(exchange.getIn()).thenReturn(message);
         when(message.getHeader(eq(FCREPO_URI))).thenReturn(BINARY_URI);
+        when(message.getHeader(eq("CamelFcrepoUri"))).thenReturn(RESC_ID);
```

Comment:
This is overriding the header from the previous line, so you only need one of these. I'd suggest updating line 73 to use FCREPO_URI and removing line 72. I don't think `BINARY_URI` will be needed anymore after you remove the duplicate line.